integration-tests/tests/basic-happy-path.spec.ts

```diff
@@ -19,7 +19,18 @@ describe('Basic Happy Path', { tags: ['@PR-check', '@publicRepo'] }, () => {
   const quarkusDeplomentBody = 'Congratulations, you have created a new Quarkus cloud application';
 
   after(function () {
-    Applications.deleteApplication(applicationName);
+    // If some test failed, don't remove the app
+    let allTestsSucceeded = true;
+    if (
+      this.test.parent.eachTest((test) => {
+        if (test.state == 'failed') {
+          allTestsSucceeded = false;
+        }
+      })
+    )
```

Comment:
am I seeing this correctly, that the whole block is an if condition?

Reply:
:woman_facepalming: